src/test/java/com/github/sdonkov/PrefixTreeTest.java

```diff
@@ -1,28 +1,76 @@
 package com.github.sdonkov;
 
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
 import org.junit.jupiter.api.BeforeAll;
 import org.junit.jupiter.api.Test;
+import org.openjdk.jmh.annotations.*;
+import org.openjdk.jmh.runner.Runner;
+import org.openjdk.jmh.runner.RunnerException;
+import org.openjdk.jmh.runner.options.Options;
+import org.openjdk.jmh.runner.options.OptionsBuilder;
+import org.openjdk.jol.info.GraphLayout;
 
 
 import java.io.IOException;
 import java.net.URISyntaxException;
 import java.nio.file.Files;
 import java.nio.file.Paths;
+import java.util.HashSet;
 import java.util.List;
+import java.util.Set;
+import java.util.TreeSet;
+import java.util.concurrent.TimeUnit;
 
 import static org.junit.jupiter.api.Assertions.*;
 
-
+@BenchmarkMode(Mode.AverageTime)
+@OutputTimeUnit(TimeUnit.MILLISECONDS)
+@Fork(value = 1, warmups = 1)
+@State(Scope.Benchmark)
 public class PrefixTreeTest {
 
     private static List<String> words;
     private static PrefixTree prefixTree;
+    private final static Logger LOGGER = LogManager.getLogger(PrefixTree.class);
 
     @BeforeAll
     static void setUp() throws IOException, URISyntaxException {
         prefixTree = new PrefixTree();
-        words = Files.readAllLines(Paths.get(PrefixTreeTest.class.getResource("/tests.txt").toURI()));
+        words = Files.readAllLines(Paths.get(PrefixTreeTest.class.getResource("/dictionary.txt").toURI()));
+        words.forEach(word -> prefixTree.add(word));
+    }
+
+    @Test
+    public void launchBenchmark() throws RunnerException {
+        Options opt = new OptionsBuilder()
+                .include(PrefixTreeTest.class.getSimpleName())
+                .build();
+        new Runner(opt).run();
+    }
+
+    @Benchmark
+    public void benchmarkPrefixTree() throws URISyntaxException, IOException {
+        prefixTree = new PrefixTree();
+        words = Files.readAllLines(Paths.get(PrefixTreeTest.class.getResource("/dictionary.txt").toURI()));
         words.forEach(word -> prefixTree.add(word));
+        LOGGER.trace(GraphLayout.parseInstance(prefixTree).totalSize() + " prefixTree");
+    }
+
+    @Benchmark
+    public void benchmarkHashSet() throws URISyntaxException, IOException {
+        words = Files.readAllLines(Paths.get(PrefixTreeTest.class.getResource("/dictionary.txt").toURI()));
+        Set<String> hashSet = new HashSet<>(words);
+        LOGGER.trace(GraphLayout.parseInstance(hashSet).totalSize() + " hashSet");
+
+    }
+
+
+    @Benchmark
+    public void benchmarkTreeSet() throws URISyntaxException, IOException {
+        words = Files.readAllLines(Paths.get(PrefixTreeTest.class.getResource("/dictionary.txt").toURI()));
+        Set<String> treeSet = new TreeSet<>(words);
+        LOGGER.trace(GraphLayout.parseInstance(treeSet).totalSize() + " treeSet");
```

Comment:
Test code is like any other code and benefits from [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself) too.